src/openms/source/SYSTEM/File.cpp

```diff
@@ -140,6 +140,48 @@ namespace OpenMS
     return true;
   }
 
+  // https://stackoverflow.com/questions/2536524/copy-directory-using-qt
+  bool File::copyDirRecursively(const QString &fromDir, const QString &toDir, bool overwrite_existing)
+  {
+    QDir sourceDir(fromDir);
+    QDir targetDir(toDir);
+  
```

Comment:
check for self-assignment?! i.e. what if fromDir==toDir? (probably requires a canonical path)